### ISSUE TITLE
Job time-outs, busy resources and trailing slashes

### ISSUE THREAD

**Issue by AgedMordorBlue** [odomlab2/sci-rocket#27]
Hi Job,

I'm experiencing multiple issues with the pipeline at the moment since I installed the last update.
It started with these warnings:
```
File path andrei_annot/raw_reads//Undetermined_S0_R1_001.fastq.gz contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
File path andrei_annot/raw_reads//Undetermined_S0_R2_001.fastq.gz contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
File path andrei_annot/raw_reads//fake.csv contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
File path andrei_annot/raw_reads//Undetermined_S0_R1_001.fastq.gz contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
File path andrei_annot/raw_reads//Undetermined_S0_R2_001.fastq.gz contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
File path andrei_annot/raw_reads//fake.csv contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
```
From what I gathered, something goes wrong with assigning the sequencing_name parameter leading to a blank character. This blank character then causes the two slashes to double. It doesn't matter whether I name it in the sample sheet or whether I let the pipeline generate it.

After this warning, the pipeline does generate the expected output in the raw_reads directory, but after some point the read files get deleted leaving only the Stat and Report directories. Meanwhile, the pipeline also starts the generate_index job, but this times out and only generates a false symlink instead of the index.

In some cases, instead of timing out, I get a broken pipe: resource busy error. I have had this both for the Undetermined reads and fake.csv files.

When doing a dry-run, the pipeline is able to go to the end, and it seems the trailing slashes get fixed when the Undetermined reads get moved during the merge step.

I've tried the following things:

- Run different samples both with the same as with different sequencing data
- Re-write the config file and sample sheet using the examples provided
- Re-initiated the set-up for the slurm profile for snakemake

**Comment by J0bbie** [comment]
Hmm, I'll try a few different scenarios and see if I can replicate this issue.
Did this occur when starting from the BCL folder or from existing FASTQ files?

Could it be an extra `/` at the end of the path in the samplesheet?
E.g. `/bcl_folder/runX/` instead of `/bcl_folder/runX`

Best,

Job

**Comment by AgedMordorBlue** [comment]
I'm using the BCL folder, and removing the trailing / in the bcl path fixes the warning indeed, thanks! In the current example this is also the case
I also found the source of the error when generating the STAR index. I still was using False to set no index instead of "".  With this change it's generating the index normally. 

Incidentally, because I did the setup of my slurm profile again, I noticed/remembered that the generate_index job is sent on the default time set for the cluster (which is 5 minutes for me) causing it to time out. Interestingly, as far as I have found out, the other jobs don't seem to need the default time to be changed. It's fairly easy to set a long default time if you know how, but this does go at a cost of job scheduling efficiency. I was wondering if it is feasible to either add a time to the jobs when they get scheduled, or briefly mention on the setup page that this is a key setting (for slurm users).

Best,
Yani
